app/components/contest-page/leaderboard-card.ts

```diff
@@ -20,27 +20,31 @@ interface Signature {
 export default class ContestPageLeaderboardCardComponent extends Component<Signature> {
   @service declare authenticator: AuthenticatorService;
 
+  get filteredTopEntries() {
```

Comment:
@VasylMarchuk in what case would `topEntries` contain an entry that doesn't belong to this leaderboard?

Reply:
Because `topEntries` returned by the route is the result of `findAll`, which returns **all entries** currently in the store, **unfiltered**. The result of `findAll` is also a magic `Proxy` which can be updated later (background refresh), glimmer understands this update and re-triggers the filtering getter. I can move the filtering getter from the component into the controller if you'd like, so that the component receives already filtered topEntries.

But notice that all other things returned from the model hook (languages, allContests) are also unfiltered.

Reply:
In other words, there's a difference:

```ts
model () {
  return this.store.findAll(...); // <-- RecordArray, background refresh handled by Glimmer
  return this.store.findAll(...).filter(); // <-- Array, background refresh will not be reflected by the template
}
```

So the model hook ideally needs to return `RecordArray`s, plain results of `findAll`, which then need to be filtered in controller/component.

Reply:
I've moved `filteredTopEntries` from `LeaderboardCardComponent` into `ContestController` (renamed as `filteredTopLeaderboardEntries`)